libs/platform/user/ebpf_platform_user.cpp

```diff
@@ -1146,6 +1146,52 @@ ebpf_platform_thread_id()
     return GetCurrentThreadId();
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL) _Must_inspect_result_ ebpf_result_t
+    ebpf_platform_get_authentication_id(_Out_ uint64_t* authentication_id)
+{
+    ebpf_result_t return_value = EBPF_SUCCESS;
+    uint32_t error;
+    TOKEN_GROUPS_AND_PRIVILEGES* privileges = nullptr;
+    uint32_t size = 0;
+    HANDLE process_handle;
+    HANDLE token_handle;
+
+    process_handle = OpenProcess(PROCESS_QUERY_INFORMATION, false, ebpf_platform_process_id());
+    if (process_handle == nullptr) {
+        return win32_error_code_to_ebpf_result(GetLastError());
+    }
+
+    bool result = OpenProcessToken(process_handle, TOKEN_QUERY, &token_handle);
+    if (result == false) {
+        return win32_error_code_to_ebpf_result(GetLastError());
+    }
+
+    result = GetTokenInformation(token_handle, TokenGroupsAndPrivileges, nullptr, 0, (PDWORD)&size);
+    ebpf_assert(result == false);
```

Comment:
Seems inconsistent with 1183-1184.  Why bother asserting when we can hande the failure?